### PR TITLE
completing a swap navigates back to previous screen

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes APP-####

## What changed (plus any additional context for devs)

edge case if the users previous screen was an Expanded Sheet, navigate to the Wallet Screen instead of back to the Expanded Sheet

## Screen recordings / screens
pow

https://github.com/user-attachments/assets/4446ab8d-ab87-4814-a0cf-7970f76f6a23

### CODE REVIEW

File: src/navigation/Navigation.js [hunk 1]
```diff
@@ -129,6 +129,7 @@ function setTopLevelNavigator(navigatorRef) {
```
Comment (greg-schrammel) [inline src/navigation/Navigation.js:129]: vote to migrate this to ts since Im here

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1810-after-completing-a-swap-navigate-to-the-screen-user-was/34b265faf0e3860b4eae0b86c3e57bb9f042ac95.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1810-after-completing-a-swap-navigate-to-the-screen-user-was/34b265faf0e3860b4eae0b86c3e57bb9f042ac95.ipa) for 34b265faf0e3860b4eae0b86c3e57bb9f042ac95

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx [hunk 2]
```diff
@@ -329,14 +329,21 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
       clearCustomGasSettings(chainId);
       NotificationManager?.postNotification('rapCompleted');
       performanceTracking.getState().executeFn({
-        fn: Navigation.handleAction,
+        fn: () => {
+          const { routes, index } = Navigation.getState();
+          if (routes[index - 1].name === 'ExpandedAssetSheet') {
```
Comment (walmat) [inline src/__swaps__/screens/Swap/providers/swap-provider.tsx:334]:
```suggestion
          if (routes[index - 1].name === Routes.EXPANDED_ASSET_SHEET) {
```

**Review by walmat — APPROVED** [review]
did you verify this works on Android as well? I remember some funkiness going on with Android navigation not too long ago. Otherwise lgtm 👍🏼

File: src/__swaps__/screens/Swap/providers/swap-provider.tsx (hunk 2 above)
Comment (benisgold) [reply src/__swaps__/screens/Swap/providers/swap-provider.tsx:334]: this

**Review by benisgold — APPROVED** [review]

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1810-after-completing-a-swap-navigate-to-the-screen-user-was/ec2e41e37b1c0733c72326170962a63d55460212.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1810-after-completing-a-swap-navigate-to-the-screen-user-was/ec2e41e37b1c0733c72326170962a63d55460212.ipa) for ec2e41e37b1c0733c72326170962a63d55460212

**Pull request merged**